apps/test-app/frontend/src/components/tree-widget/stateless/UnifiedSelectionTreeEventHandler.ts

```diff
@@ -0,0 +1,258 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Observable as RxjsObservable, Subject, takeUntil, tap } from "rxjs";
+import {
+  AbstractTreeNodeLoader,
+  MutableTreeModel,
+  MutableTreeModelNode,
+  Observable,
+  TreeEventHandler,
+  TreeModelChanges,
+  TreeModelSource,
+  TreeNodeItem,
+  TreeSelectionModificationEventArgs,
+  TreeSelectionReplacementEventArgs,
+} from "@itwin/components-react";
+import { Guid, IDisposable } from "@itwin/core-bentley";
+import { IModelConnection } from "@itwin/core-frontend";
+import { GroupingNodeKey, Key, KeySet, PresentationQuery, PresentationQueryBinding, StandardNodeTypes } from "@itwin/presentation-common";
+import { Presentation, SelectionChangeEventArgs, SelectionChangeType, SelectionHandler } from "@itwin/presentation-frontend";
+import { HierarchyNode, InstanceKey, parseFullClassName } from "@itwin/presentation-hierarchy-builder";
+import { getHierarchyNode } from "./TreeNodeItemUtils";
+
+export interface UnifiedSelectionTreeEventHandlerParams {
+  /** The iModel whose data is being displayed in the component. */
+  imodel: IModelConnection;
+
+  /** Node loader used to load children when node is expanded. */
+  nodeLoader: AbstractTreeNodeLoader;
+
+  /** Specifies whether children should be disposed when parent node is collapsed or not. */
+  collapsedChildrenDisposalEnabled?: boolean;
+}
+
+export class UnifiedSelectionTreeEventHandler extends TreeEventHandler implements IDisposable {
```

Comment:
It looks like it should be enough to extend `[UnifiedSelectionTreeEventHandler](https://github.com/iTwin/presentation/blob/9d0bad0c74e37d07401b418779abcd0c4c63c5ac/packages/components/src/presentation-components/tree/controlled/UseUnifiedSelection.ts#L68)` and override `createKeysForSelection`, `shouldSelectNode` methods instead of copying whole implementation.

Reply:
The primary reason I didn't reuse [UnifiedSelectionTreeEventHandler](https://github.com/iTwin/presentation/blob/9d0bad0c74e37d07401b418779abcd0c4c63c5ac/packages/components/src/presentation-components/tree/controlled/UseUnifiedSelection.ts#L68) here is that it wants an `AbstractTreeNodeLoaderWithProvider<IPresentationTreeDataProvider>`, which we don't have. We could extract common behavior into a base class and share it, I don't think that's worth the effort, having in mind that all tree-related APIs for stateless will need to written from scratch in components-agnostic way.